hardware_interface/include/hardware_interface/components/joint.hpp

```diff
@@ -58,15 +58,15 @@ class Joint
    * \return string list with command interfaces.
    */
   HARDWARE_INTERFACE_PUBLIC
-  std::vector<std::string> get_command_interfaces() const;
+  std::vector<InterfaceInfo> get_command_interfaces() const;
```

Comment:
Puh... I didn't image to change this, but it is very good idea :+1: 

I can imagine that a controller need also limit-data of an interface and for that case is this function very useful. Nevertheless, we need very often interface names of a joint or sensor. Therefore, I propose the following:

```suggestion
  std::vector<std::string> get_command_interface_names() const;
  
  /**
   * \brief Provide the list of command interfaces configured for the joint.
   *
   * \return list with informations about command interfaces.
   */
  HARDWARE_INTERFACE_PUBLIC
  std::vector<InterfaceInfo> get_command_interfaces() const;
```